client/layout/masterbar/notifications.jsx

```diff
@@ -23,32 +30,51 @@ export default React.createClass( {
 		tooltip: React.PropTypes.string,
 	},
 
+	componentDidMount() {
+		wpcom
+			.req
+			.get( {
+				path: '/notifications/',
+				apiVersion: '1.1'
+			}, { number: 100 } )
+			.then( response => {
+				const { notes } = fromApi( response );
+				console.log( notes[0] );
+
+				this.setState( { notes } );
+			} );
+	},
+
 	getInitialState() {
 		let user = this.props.user.get();
 
 		return {
 			isShowingPopover: false,
 			newNote: user && user.has_unseen_notes,
 			animationState: 0,
+			notes: [],
+			selectedNote: null,
+			selectedFilter: 'All'
 		};
 	},
 
-	checkToggleNotes( event, forceToggle ) {
-		const target = event ? event.target : false;
-		const notificationNode = this.getNotificationLinkDomNode();
-
-		if ( target && notificationNode.contains( target ) ) {
-			return;
-		}
+	selectNote( selectedNote ) {
+		this.setState( { selectedNote } );
+	},
 
-		if ( this.state.isShowingPopover || forceToggle === true ) {
-			this.toggleNotesFrame( event );
-		}
+	selectFilter( selectedFilter ) {
+		this.setState( { selectedFilter } );
 	},
 
 	toggleNotesFrame( event ) {
 		if ( event ) {
 			event.preventDefault();
+
+			const notesPanel = document.getElementById( 'wpnt-notes-panel2' );
+			if ( notesPanel && notesPanel.contains( event.target ) ) {
+				event.stopPropagation();
```

Comment:
I'm curious about the problem here is.  It looks like the notifications are under the masterbar, and I'm missing whatever it is that the event would bubble up to to make problems.